hack/generate-schemes/main.go

```diff
@@ -0,0 +1,175 @@
+// SPDX-FileCopyrightText: 2021 "SAP SE or an SAP affiliate company and Gardener contributors"
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/gardener/landscaper/apis/hack/generate-schemes/generators"
+	lsschema "github.com/gardener/landscaper/apis/schema"
+	"github.com/go-openapi/spec"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"sigs.k8s.io/yaml"
+
+	corelsv1alpha1 "github.com/gardener/landscaper-service/pkg/apis/core/v1alpha1"
+
+	"github.com/go-openapi/jsonreference"
+
+	"github.com/gardener/landscaper-service/pkg/apis/openapi"
+)
+
+var Exports = []string{
+	"LandscaperDeployment",
+	"Instance",
+	"SeedConfig",
+}
+
+var CRDs = []lsschema.CustomResourceDefinitions{
+	corelsv1alpha1.ResourceDefinition,
+}
+
+var (
+	schemaDir string
+	crdDir    string
+)
+
+func init() {
+	flag.StringVar(&schemaDir, "schema-dir", "", "output directory for jsonschemas")
+	flag.StringVar(&crdDir, "crd-dir", "", "output directory for crds")
+}
+
+func main() {
+	flag.Parse()
+	if len(schemaDir) == 0 {
+		log.Fatalln("expected --schema-dir to be set")
+	}
+	if err := run(schemaDir, crdDir); err != nil {
+		fmt.Println(err.Error())
+		os.Exit(1)
+	}
+}
+
+func run(schemaDir, crdDir string) error {
```

Comment:
seems like there is still a lot of copied code.
Maybe this is also something we could share and put into a lib.

Reply:
It certainly is.
Would this be a candidate for https://github.com/gardener/landscaper-utils ?

Reply:
maybe. But landscaper-utils was more meant for utils regarding building blueprint or interacting with the landscaper.
This is more a generic crd generation approach.

So I would rather include it to the generator in the landscaper-api repo.
Otherwise the landscaper would have to depend on the landscaper-utils if we do not want to copy code again.

Reply:
Ok, that makes sense.
However, I would like to open a seperate issue/pr in landscaper and use this as a temporary solution until it is available in the landscaper api module.